src/pipeline.cc

```diff
@@ -76,19 +77,25 @@ using sharp::FreeCallback;
 using sharp::CalculateCrop;
 using sharp::counterProcess;
 using sharp::counterQueue;
+using sharp::GetBooleanOperation;
+
+typedef struct BufferContainer_t {
+  std::string name;
+  Local<Object> nodeBuf;
+  size_t length;
+} BufferContainer;
 
 class PipelineWorker : public AsyncWorker {
  public:
   PipelineWorker(Callback *callback, PipelineBaton *baton, Callback *queueListener,
-    const Local<Object> &bufferIn, const Local<Object> &overlayBufferIn) :
+                 const std::vector<BufferContainer> saveBuffers) :
     AsyncWorker(callback), baton(baton), queueListener(queueListener) {
-      if (baton->bufferInLength > 0) {
-        SaveToPersistent("bufferIn", bufferIn);
-      }
-      if (baton->overlayBufferInLength > 0) {
-        SaveToPersistent("overlayBufferIn", overlayBufferIn);
+    for (const BufferContainer b : saveBuffers) {
+      if (b.length > 0) {
+        SaveToPersistent(b.name.c_str(), b.nodeBuf);
```

Comment:
We can probably auto-generate a suitable "key" (the first parameter, a string) for `SaveToPersistent` based on the memory location, perhaps something like http://stackoverflow.com/a/7850160/4625124